### PR TITLE
palemoon: 29.4.4 -> 30.0.0

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

https://www.palemoon.org/releasenotes.shtml

> Updated various in-tree libraries: cubeb, sqlite, cairo, ...
> Fixed an issue with the Linux desktop shortcut file to solve potential DE integration problems on common distributions.
> Fixed an issue with page and iframe content margins not being applied properly when passed as attributes instead of CSS.
> Ensured JavaScript and JSON files are always recognized as known MIME types so they can be opened appropriately from local sources.
> Fixed an issue with rapid loading and unloading of js modules causing browser crashes.
> Fixed an issue with tooltips being cut off at the end if containing exceedingly long unwrappable series of characters.
> Fixed several application crash scenarios. DiD
> Fixed a large number of thread locking/mutex issues. DiD
> Fixed a leak of content types due to inconsistent error reporting. (CVE-2022-22760)
> Fixed an issue with iframe sandboxing not being properly applied. (CVE-2022-22759)
> Fixed a potential leak of bookmarks from the exported bookmarks file if it included a malicious bookmarklet.
> Fixed an issue with drag-and-drop. (CVE-2022-22756)
> Fixed a potential crash due to truncated WAV files.
> Fixed a memory safety issue with XSLT. (CVE-2022-26485)

WIP. `mach build` & `mach install` complete but the rest of `installPhase` needs to be updated.

```console
 1:59.39(B '/nix/store/by69zqgj36k242p9xkwr4l6pwfx6hlsf-palemoon-30.0.0/bin/palemoon' -> '/nix/store/by69zqgj36k242p9xkwr4l6pwfx6hlsf-palemoon-30.0.0/lib/palemoon-30.0.0/palemoon'
 1:59.39(B To run the installed application, execute: .//nix/store/by69zqgj36k242p9xkwr4l6pwfx6hlsf-palemoon-30.0.0/bin/palemoon .
 1:59.39(B make[1]: Leaving directory '/build/source/obj-x86_64-pc-linux-gnu/browser/installer'
 1:59.39(B make: Leaving directory '/build/source/obj-x86_64-pc-linux-gnu'
substitute(): ERROR: file './palemoon/branding/official/palemoon.desktop' does not exist
builder for '/nix/store/d36gz3wkl6lygkqlzdr588w7ggfid6fl-palemoon-30.0.0.drv' failed with exit code 1
error: build of '/nix/store/d36gz3wkl6lygkqlzdr588w7ggfid6fl-palemoon-30.0.0.drv' failed
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
After figuring out where the branding was moved to & what workarounds upstream deprecated, it seems to work.

Bildschirmfoto von 2022-03-19 19-01-50

**Comment by OPNA2608** [comment]
@ofborg eval

File: pkgs/applications/networking/browsers/palemoon/default.nix [hunk 1]
```diff
@@ -137,24 +140,15 @@ stdenv.mkDerivation rec {
...
     desktop-file-install --dir=$out/share/applications \
-      ./palemoon/branding/official/palemoon.desktop
-
-    # Install official branding icons
+      ./other-licenses/branding/palemoon/official/palemoon.desktop
     for iconname in default{16,22,24,32,48,256} mozicon128; do
       n=''${iconname//[^0-9]/}
       size=$n"x"$n
-      install -Dm644 ./palemoon/branding/official/$iconname.png $out/share/icons/hicolor/$size/apps/palemoon.png
+      install -Dm644 ./other-licenses/branding/palemoon/official/$iconname.png $out/share/icons/hicolor/$size/apps/palemoon.png
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/palemoon/default.nix:149]: What does `other-licenses` mean here?
Yes, it is just a directory name, but is there something we should pay attention here?

Comment (OPNA2608) [reply pkgs/applications/networking/browsers/palemoon/default.nix:149]: All branding ("official", "beta", "unstable" etc) except the MPL2.0-licensed "unofficial" one (which is included for building convenience and intended to be replaced for distribution) has been moved to `other-licenses/branding/palemoon` in this major release.

The licensing situation itself is unchanged: The branding is (and always have been) under a [non-MPL2.0 license](https://repo.palemoon.org/MoonchildProductions/Pale-Moon/src/branch/master/other-licenses/branding/palemoon/LICENSE) that gives Moonchild Productions exclusive ownership of the branding & "Pale Moon" name. [This post](https://forum.palemoon.org/viewtopic.php?f=24&t=26031) outlines when the official branding may be used for distributed source builds.

tl;dr just the same as always w/r/t permitted compilers, mozconfig settings, use of in-tree libraries, no modifications to defaults etc.

Comment (AndersonTorres) [reply pkgs/applications/networking/browsers/palemoon/default.nix:149]: OK then!

**Comment by OPNA2608** [comment]
After the infra outage, the v30 milestone was recalled: https://forum.palemoon.org/viewtopic.php?f=1&t=28044.

Bildschirmfoto von 2022-03-22 07-57-52

I think we need to revert this.
